apps/rehash.c

```diff
@@ -356,10 +353,10 @@ static int do_dir(const char *dirname, enum Hash h)
     struct stat st;
     unsigned char idmask[MAX_COLLISIONS / 8];
     int n, numfiles, nextid, dirlen, buflen, errs = 0;
-    size_t i;
+    size_t i, fname_max_len = 20; /* maximum length of "%08x.r%d" */
```

Comment:
I see from the comment where this minimal size is derived from, but immediately below line 425 there is this code:
```
BIO_snprintf(buf, buflen, "%s%s%08x.%s%d",
                                 dirname, pathsep, bp->hash,
                                 suffixes[bp->type], nextid);
```

If the target dirname is short (say its just ./) and the file names are likewise short, it seems that fname_max_length will never get increased, but be insufficient for a maximially sized hash/type/id tuple

Reply:
The dirname and pathsep is accounted for separately. The maximum length of %08x.%s%d is 20 (type is at most one character). In reality it will be much shorter as the nextid value will be a small number.